store/v2/storage/sqlite/db.go

```diff
@@ -203,14 +204,28 @@ func (db *Database) Prune(version uint64) error {
 	) AND store_key != ?;
 	`
 
-	_, err = tx.Exec(pruneStmt, version, reservedStoreKey)
-	if err != nil {
+	if _, err = tx.Exec(pruneStmt, version, reservedStoreKey); err != nil {
+		return fmt.Errorf("failed to exec SQL statement: %w", err)
+	}
+
+	// prune removed store keys
+	pruneRemovedStoreKeysStmt := `DELETE FROM state_storage
+	WHERE store_key in (
```

Comment:
not fully sure if I understood the DB schema but I think that you also have to define the version to not delete all for a storage key